daemon/dns/dns_resolvconf_file.go

```diff
@@ -94,3 +101,86 @@ func restoreDNS() error {
 	}
 	return internal.FileWrite(resolvconfFilePath, out, internal.PermUserRWGroupROthersR)
 }
+
+func tryToRestoreDNS() {
+	// if target is symlink, probably it is managed by other software - do nothing
+	if internal.IsSymLink(resolvconfFilePath) {
+		return
+	}
+	// if target /etc/resolv.conf contains nordvpn changes:
+	// if possible restore from backup
+	// if backup does not exists, create simple dns settings file
+	out, err := internal.FileRead(resolvconfFilePath)
+	if err != nil {
+		log.Println(internal.ErrorPrefix, fmt.Errorf("reading resolv.conf: %w", err))
+		return
+	}
+	if !strings.Contains(string(out), resolvconfFileMark) {
+		return
+	}
+
+	log.Println(internal.WarningPrefix, "/etc/resolv.conf contains our changes - need to fix this")
+
+	// try to unlock, if file was locked
+	_ = internal.FileUnlock(resolvconfFilePath)
+	// restore from backup if backup file exists
+	if internal.FileExists(resolvconfBackupPath) {
+		log.Println(internal.WarningPrefix, "resolv.conf backup found - restore from it")
+		backup, err := internal.FileRead(resolvconfBackupPath)
+		if err != nil {
+			log.Println(internal.ErrorPrefix, fmt.Errorf("reading backup resolv.conf: %w", err))
+		} else {
+			if !strings.Contains(string(out), resolvconfFileMark) {
+				if err := internal.FileWrite(resolvconfFilePath, backup, internal.PermUserRWGroupROthersR); err != nil {
+					log.Println(internal.ErrorPrefix, fmt.Errorf("restore from backup resolv.conf: %w", err))
+				}
+			}
+		}
+		// remove backup
+		internal.FileDelete(resolvconfBackupPath)
+	}
+
+	log.Println(internal.WarningPrefix, "/etc/resolv.conf contains our changes - restore with simple dns settings")
+
+	// there is no backup, but we need to fix dns settings
+	ip, err := discoverNameserverIp()
+	if err != nil {
+		log.Println(internal.ErrorPrefix, fmt.Errorf("discovering nameserver: %w", err))
+		// this is very-very last option and hope this will not happen
+		ip = netip.MustParseAddr("1.1.1.1")
+	}
+
+	log.Println(internal.WarningPrefix, "/etc/resolv.conf restore with nameserver:", ip)
+
+	content := fmt.Sprintf(resolvconfFileContent, ip)
+	if err := internal.FileWrite(resolvconfFilePath, []byte(content), internal.PermUserRWGroupROthersR); err != nil {
+		log.Println(internal.ErrorPrefix, fmt.Errorf("writing simple resolv.conf: %w", err))
+	}
+}
+
+func discoverNameserverIp() (netip.Addr, error) {
+	// 1st try: 127.0.0.53:53
+	// then: GW
+	nsIp := netip.MustParseAddr("127.0.0.53")
+	_, err := net.ListenPacket("udp", nsIp.String()+":53")
```

Comment:
The connection `_` needs to be closed to stop listening on that port when there is no error returned.

Reply:
No connection is being created - it's UDP

Reply:
But it opens a socket, puts it into a listen state and returns the connection object. We need to call close on that object to stop listening on the socket.

Reply:
No object is created - variable is `nil`. Ok, looking into that more

Reply:
According to the documentation it should return the connection object when there were no errors.

Reply:
Yes, you are right, added closing the connection